include/CL/sycl/parallelism/detail/parallelism.hpp

```diff
@@ -182,12 +186,22 @@ void parallel_for(range<Dimensions> r,
     index type of the kernel function object f
 
 */
+#if !defined(TRISYCL_USE_OPENCL_ND_RANGE)
 template <int Dimensions = 1, typename ParallelForFunctor>
 void parallel_for(range<Dimensions> r, ParallelForFunctor f) {
   using mf_t  = decltype(std::mem_fn(&ParallelForFunctor::operator()));
   using arg_t = typename mf_t::second_argument_type;
   parallel_for(r,f,arg_t{});
 }
+#else
+template <int Dimensions = 1, typename ParallelForFunctor>
+void parallel_for(range<Dimensions> r, ParallelForFunctor f) {
+  using mf_t  = decltype(std::mem_fn(&ParallelForFunctor::operator()));
```

Comment:
Very clever hack that would require some comments. But at the same time `std::mem_fn` is deprecated in C++17 so at some point we will need another solution... Create an issue to track this?

Reply:
Actually I wonder whether all this could not be simplified by another commit adding the equivalent of `arg_t` as a third template parameter to `parallel_for` so we do not have to recover it...

Reply:
It's actually not deprecated in C++17 I believe you are referring to: https://en.cppreference.com/w/cpp/utility/functional/mem_fun however, in this case we're using https://en.cppreference.com/w/cpp/utility/functional/mem_fn which gets a noexcept operator added in C++17. 

It replaced mem_fun as it's more generic! 1 letter difference for a replacement function is a nice way for the C++ committee to confuse people I guess, but allows for easier replacement I suppose.